### PR TITLE
BCDA-4533 Update resource types for ALR 

### PR DESCRIPTION
**Pull request by CosmicSyntax**
### Fixes [BCDA-4533](https://jira.cms.gov/browse/BCDA-4533)
This PR is to match updates Bill Hardy has provided on more stable and possibly permanent FHIR homes for variables in table 1-1 to 1-4 in ALR.
### Proposed Changes
Previously, we were outputting ALR data using only two FHIR resource types: patient and observation. Also, the output of the data was all put into one file, which is not correct. We add new FHIR resources, and a separate file for each FHIR resource.
### Change Details
We added additional FHIR resource types: coverage, group, risk assessment. This effort to align closely with the Da Vinci working group that is working on something very similar to ALR.
Also, there is now a separate ndjson file for each FHIR resource type.
### Security Implications
- [ ] new software dependencies
- [ ] security controls or supporting software altered
- [X] new data stored or transmitted
- [ ] security checklist is completed for this change
- [ ] requires more information or team discussion to evaluate security implications
- [X] no PHI/PII is affected by this change

ALR data (synthetic) should now be contained in 5 FHIR resources: patient, coverage, group, risk assessment, and observation.
### Acceptance Validation
Update looks good and all test passes.

### Feedback Requested
This is fairly a large PR and this first of implement is nowhere near perfect. Additional tickets may be needed to be ready for production,

### CODE REVIEW

File: bcda/models/fhir/alr/alr.go [hunk 1]
```diff
@@ -1,22 +1,40 @@
...
+	Coverage    *fhirmodels.Coverage
+	Group       *fhirmodels.Group
+	Risk        []*fhirmodels.RiskAssessment
+	Observation *fhirmodels.Observation
+}
+
 // ToFHIR encodes the models.Alr into a FHIR Patient and N FHIR Observation resources
-func ToFHIR(alr *models.Alr, lastUpdated time.Time) (*fhirmodels.Patient, []*fhirmodels.Observation) {
-	p := patient(alr)
-	p.Meta = &fhirdatatypes.Meta{LastUpdated: fhirInstant(lastUpdated)}
```
Comment (katyasoup) [inline bcda/models/fhir/alr/alr.go:14]: Why did we drop the last_updated field in the ALR metadata? For non-ALR I beleive we return it with the Bundle resource, should we include it as part of AlrFhirBulk?

File: bcda/models/fhir/alr/alr_test.go [hunk 2]
```diff
@@ -34,56 +37,92 @@ func TestGenerateAlr(t *testing.T) {
 	assert.NoError(t, err)
 	assert.Len(t, alrs, 1)
 
-	lastUpdated := time.Now().Round(time.Second)
-	patient, obs := alr.ToFHIR(alrs[0], lastUpdated)
-	assert.NotNil(t, patient)
-	assert.Len(t, obs, 5)
+	//lastUpdated := time.Now().Round(time.Second)
```
Comment (katyasoup) [inline bcda/models/fhir/alr/alr_test.go:40]: remove if dropping lastUpdated, or include as part of AlrBulkFhir struct

File: bcda/models/fhir/alr/coverage.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
+package alr
+
+import (
+	//"github.com/CMSgov/bcda-app/bcda/models"
+	//fhircodes "github.com/google/fhir/go/proto/google/fhir/proto/stu3/codes_go_proto"
```
Comment (katyasoup) [inline bcda/models/fhir/alr/coverage.go:5]: nitpicking uneeded comments :)
also, confirming that r4 update will be in a separate pr?

File: bcda/models/fhir/alr/types.go [hunk 4]
```diff
@@ -23,10 +23,10 @@ func fhirDateTime(t time.Time) *fhirdatatypes.DateTime {
 	return &fhirdatatypes.DateTime{ValueUs: micros, Precision: fhirdatatypes.DateTime_DAY}
 }
 
-func fhirInstant(t time.Time) *fhirdatatypes.Instant {
-	micros := t.UnixNano() / int64(time.Microsecond)
-	return &fhirdatatypes.Instant{ValueUs: micros, Precision: fhirdatatypes.Instant_SECOND}
-}
+//func fhirInstant(t time.Time) *fhirdatatypes.Instant {
```
Comment (katyasoup) [inline bcda/models/fhir/alr/types.go:26]: can this be deleted?

File: bcdaworker/queueing/manager/que_test.go [hunk 5]
```diff
@@ -193,12 +193,13 @@ func TestStartAlrJob(t *testing.T) {
...
 
 	// just use one mbi for this test
-	twoMbis := make([]string, 2)
+	twoMbis := []string{}
 	twoMbis = append(twoMbis, mbis[0].MBI)
 	twoMbis = append(twoMbis, mbis[1].MBI)
 
 	alr, err := alrWorker.GetAlr(ctx, *aco.CMSID, twoMbis, time.Time{}, time.Time{})
 	assert.NoError(t, err)
+	fmt.Println("!!!", alr)
```
Comment (katyasoup) [inline bcdaworker/queueing/manager/que_test.go:202]: remove println :)

File: bcdaworker/worker/alr.go [hunk 6]
```diff
@@ -158,38 +222,40 @@ func (a *AlrWorker) ProcessAlrJob(
...
 	// Reason for a go routine is to not block when writing, since disk writing is
 	// generally slower than memory access. We are streaming to keep mem lower.
-	go goWriter(c, w, marshaller, result)
+	go goWriter(ctx, a, c, fileMap, marshaller, result, resources[:], id)
 
 	// Marshall into JSON and send it over the channel
 	for i := range alrModels {
-		patient, observations := alr.ToFHIR(&alrModels[i], alrModels[i].Timestamp)
-		c <- data{patient, observations}
+		fhirBulk := alr.ToFHIR(&alrModels[i]) // Removed timestamp, but can be added back here
```
Comment (katyasoup) [inline bcdaworker/worker/alr.go:235]: I'd consider adding it back, we include this with non-ALR resources

**Review by katyasoup — COMMENTED** [review]
Looks good overall! Mostly nitpicking about tidying up comments, but I am curious about dropping the last_updated from the metadata

File: bcda/models/fhir/alr/alr.go (hunk 1 above)
Comment (CosmicSyntax) [reply bcda/models/fhir/alr/alr.go:14]: I don't think I saw it in the implementation guide, but you're right. Let's update this back in another ticket.

File: bcdaworker/worker/alr.go (hunk 6 above)
Comment (CosmicSyntax) [reply bcdaworker/worker/alr.go:235]: I think so too. I'll do this in the other ticket that implements r4.

File: bcda/models/fhir/alr/alr_test.go (hunk 2 above)
Comment (CosmicSyntax) [reply bcda/models/fhir/alr/alr_test.go:40]: Will add this back in in R4 ticket.

**Review by SMLuthi — APPROVED** [review]
LGTM! Besides the things Katie pointed out I think this looks solid!

**Review by katyasoup — APPROVED** [review]
🎉
